extensions/script-libraries/drawing/drawing.livecodescript

```diff
@@ -0,0 +1,2960 @@
+		case "fill-opacity"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeFillOpacity, tArgument
+			break
+		case "fill-rule"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeFillRule
+			if tArgument is "nonzero" then
+				_svgEncodeOp xContext, kMCGDrawingFillRuleOpcodeNonZero
+			else if tArgument is "evenodd" then
+				_svgEncodeOp xContext, kMCGDrawingFillRuleOpcodeEvenOdd
+			else
+				_InternalError format("unknown fill-rule '%s'", tArgument)
+			end if
+			break
+		case "stroke"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeStrokePaint
+			_svgEncodePaint xContext, tArgument
+			break
+		case "stroke-opacity"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeStrokeOpacity, tArgument
+			break
+		case "stroke-width"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeStrokeWidth, tArgument
+			break
+		case "stroke-linejoin"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeStrokeLineJoin
+			switch tArgument
+			case "bevel"
+				_svgEncodeOp xContext, kMCGDrawingStrokeLineJoinOpcodeBevel
+				break
+			case "round"
+				_svgEncodeOp xContext, kMCGDrawingStrokeLineJoinOpcodeRound
+				break
+			case "miter"
+				_svgEncodeOp xContext, kMCGDrawingStrokeLineJoinOpcodeMiter
+				break
+			end switch
+			break
+		case "stroke-linecap"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeStrokeLineCap
+			switch tArgument
+			case "butt"
+				_svgEncodeOp xContext, kMCGDrawingStrokeLineCapOpcodeButt
+				break
+			case "round"
+				_svgEncodeOp xContext, kMCGDrawingStrokeLineCapOpcodeRound
+				break
+			case "square"
+				_svgEncodeOp xContext, kMCGDrawingStrokeLineCapOpcodeSquare
+				break
+			end switch
+			break
+		case "stroke-dasharray"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeStrokeDashArray, tArgument
+			break
+		case "stroke-dashoffset"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeStrokeDashOffset, tArgument
+			break
+		case "stroke-miterlimit"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeStrokeMiterLimit, tArgument
+			break
+		case "rect"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeRectangle, tArgument[1], tArgument[2], tArgument[3], tArgument[4], tArgument[5], tArgument[6]
+			break
+		case "circle"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeCircle, tArgument[1], tArgument[2], tArgument[3]
+			break
+		case "ellipse"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeEllipse, tArgument[1], tArgument[2], tArgument[3], tArgument[4]
+			break
+		case "line"
+			_svgEncodeOp xContext, kMCGDrawingOpcodeLine, tArgument[1], tArgument[2], tArgument[3], tArgument[4]
+			break
+		case "polyline"
+			_svgEncodeOp xContext, kMCGDrawingOpcodePolyline, tArgument
+			break
+		case "polygon"
+			_svgEncodeOp xContext, kMCGDrawingOpcodePolygon, tArgument
+			break
+		case "path"
+			_svgEncodeOp xContext, kMCGDrawingOpcodePath
+			_svgEncodePath xContext, tArgument
+			break
+		default
+			_InternalError format("unknown command '%s'", tOperation[1])
+			break
+		end switch
+	end repeat
+
+	_svgEncodeOp xContext, kMCGDrawingOpcodeEnd
+
+	local tDrawing
+	put "LCD" & numToByte(0) into tDrawing
```

Comment:
I think this `0` should be a version constant, probably somewhere near the top of the script

Reply:
Sure - I've changed that to use binaryEncode with appropriate constants.